src/tables.jl

```diff
@@ -115,3 +115,65 @@ end
 
 #     end
 # end
+
+"""
+    populate!(A, table, value)
+
+Populate A with the contents of the `value` column in a provided table.
+The provided table contain columns corresponding to the keys in A and support row iteration.
+"""
+function populate!(A, table, value::Symbol)
+    cols = [value, dimnames(A)...]
+    for r in Tables.rows(table)
+        setkey!(A, (Tables.getcolumn(r, c) for c in cols)...)
```

Comment:
I'm not sure if there's a faster way to do this, but most of the time is spent on this one line. I'm guessing it's just cause we're doing a linear search every time, so outside of memoizing the lookup I'm guessing this the best we can do?

Reply:
I guess for 3 columns & N rows, this does N^3 lookups, while memoizing would do only 3N, plus 3N^3 hashes. 

That seems fine to me. If you want it fast, then should make the key-vectors UniqueVector or something, and you're back to 3N^3 hashes. Or if the key-vectors are AbstractRanges, then lookup should be even quicker.

Reply:
I played around a little with AcceleratedArray and UniqueVectors a bit and they didn't seem to make a big difference to performance... at least with the example in the tests.

Reply:
In the similar `freqtable` code in FreqTables.jl, I have optimized methods for `PooledArray` and `CategoricalArray` (using integer refs for indexing), which are quite common for this kind of use case I guess. Could be worth adding later.

Also, does the `(Tables.getcolumn(r, c) for c in cols)` part take a significant part of the time? If so, I think you can get something much faster for type-unstable structures like `DataFrame` by passing a `Tables.namedtupleiterator` to an inner function (e.g. using `foreach`). Then you don't have to lookup the names for each row -- but you pay the price of compilation for each set of column names and types, so it may not be worth it.